src/generic-registry.spec.ts

```diff
@@ -0,0 +1,117 @@
+// import { Observable } from 'rxjs'
+import { Fish, FishId, Pond, Tag } from '@actyx/pond'
+import { Observable } from 'rxjs'
+import { take } from 'rxjs/operators'
+import { FishName, generic, RegisterableFish } from './index'
+
+type Event =
+    | {
+        fish: string
+        foo: number
+    }
+    | {
+        fish: string
+        foo: string
+        bar: number
+    }
+    | string
+
+type State = string
+
+const TestTag = Tag<Event>('generic-test')
+const mkTestFish = (name: FishName): Fish<State, Event> => ({
+    initialState: name,
+
+    onEvent: (state, _event) => state,
+
+    where: TestTag.withId(name),
+
+    fishId: FishId.of('test', name, 0),
+})
+
+const TestFish: RegisterableFish<FishName, State, Event> = {
+    descriptor: 'test',
+
+    makeFish: mkTestFish,
+
+    extractFishName: (e: Event) => (typeof e === 'string' ? e : e.fish),
+
+    events: TestTag,
+}
+
+const emitEvent = async (pond: Pond, e: Event) =>
+    await pond
+        .emit<Event>(TestTag.withId(TestFish.extractFishName(e)), e)
+        .toPromise()
+        .then(() => new Promise(resolve => setTimeout(resolve, 0)))
+
+const nxt = async <T>(o: Observable<T>) => await take(1)(o).toPromise()
+
+describe('generic registry', () => {
```

Comment:
could you add tests for `createRegistryFish`, too? test coverage will be huuuge!

Reply:
Well it’s used by `observeAll`, isn’t it? I don’t understand why the coverage calculator complains :(